pkg/k8s/client/object_tracker.go

```diff
@@ -0,0 +1,617 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package client
+
+// Some parts of this code is originally from k8s.io/client-go/testing
+// Copyright 2015 The Kubernetes Authors.
```

Comment:
Bunch of this code is copied from the original `testing.tracker`. Not sure how this should be attributed.